lib/tools/system-calls.js

```diff
@@ -727,20 +721,27 @@ systemCallMethods.checkAvdExist = async function (avdName) {
  * @throws {Error} If the emulator is not ready within the given timeout.
  */
 systemCallMethods.waitForEmulatorReady = async function (timeoutMs = 20000) {
-  let start = Date.now();
-  log.debug("Waiting until emulator is ready");
-  while ((Date.now() - start) < timeoutMs) {
-    try {
-      let stdout = await this.shell(["getprop", "init.svc.bootanim"]);
-      if (stdout.indexOf('stopped') > -1) {
-        return;
+  try {
+    await waitForCondition(async () => {
+      try {
+        if (!await this.shell(['getprop', 'init.svc.bootanim']).includes('stopped')) {
```

Comment:
I don't think this condition will ever work, since you are checking `includes` of the promise returned by `this.shell`. Some parentheses need to be added.

Reply:
yep, good catch